impress-org/givewp
# Enhancement: Add excerpt length option to Form Grid Block

Resolves #6323

## Description


The Form Grid block lacked the ability to set the displayed length of the form excerpt, a feature that the Form Grid shortcode has. It would always show the first 15 words of the excerpt. This PR fixes that by adding an "Excerpt Length" setting in the Form Grid Block options. The "Excerpt Length" option is only visible when "Show Excerpt" is set to true, otherwise it's hidden using `display: none;`.

## Affects


N/A

## Visuals


![image](https://user-images.githubusercontent.com/7819986/161636618-2f6d5d14-6fbf-4afc-9873-3987c76f3e5a.png)

## Testing Instructions


1. Install GiveWP
2. Create a new form or edit an existing form
3. Scroll to the bottom of the Edit Donation Form screen and add an excerpt, then click "Publish"
4. Create a new post or page and add a Form Grid block (click the '+' and search for Donation Form Grid)
5. Click on the Form Grid block and toggle "Show Excerpt" on
6. Input various values for the "Excerpt Length" option and verify that the excerpt shows that number of words
7. Publish the post/page
8. View the form grid from the frontend and verify that the excerpt shows the correct number of words

## Pre-review Checklist
